src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int SWITCH_BOUND = 5;
+    private static final int RADIUS_OF_DEFAULT_FIGURE = 10;
+    private static final int ORIGIN = 1;
+    private static final int MAX_VALUE = 100;
+    private static final String COLOR_OF_DEFAULT_FIGURE = Color.WHITE.name();
+
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+
+        int index = random.nextInt(SWITCH_BOUND);
+        String color = colorSupplier.getRandomColor();
+
+        switch (index) {
+            case 0 -> {
+                return new Circle(randomNumber(),
+                        color);
+            }
+            case 1 -> {
+                return new IsoscelesTrapezoid(randomNumber(),
+                        randomNumber(),
```

Comment:
The properties of the IsoscelesTrapezoid are not validated. The sum of the lengths of two sides of a trapezoid must always be greater than the length of the base for the area to exist.

Reply:
done